uint8 REPLY_V1 = 0
uint8 REPLY_V2 = 1
uint8 REPLY_V3 = 2

uint8 reply_mode